app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,73 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def transform
+
+      if (params[:date_coverage_1_year].length>0)
+        yyyy1= params[:date_coverage_1_year].to_i 
+      else
+        yyyy2 = 0 #error 
+      end  
+      if (!params[:date_coverage_1_month].eql?("--"))
```

Comment:
It is not so much a matter of style, but of behavior.  `eql?` and `==` operate differently for most classes.  For example `1 == 1.0` vs `1.eql? 1.0`  Most classes will override `==` when changing the behavior of comparison is required.  In many cases, `eql?` will be redirected to use `==`.